airflow/ui/src/pages/Variables/Variables.tsx

```diff
@@ -0,0 +1,260 @@
+              return !valueToFilter.startsWith(filterText);
+            case "starts_with":
+              return valueToFilter.startsWith(filterText);
+            default:
+              return true;
+          }
+        });
+      }
+
+      setFilteredData(filtered);
+    } else {
+      setFilteredData([]);
+    }
+
+    // Reset pagination when filter changes
+    setTableURLState({
+      pagination: { ...pagination, pageIndex: 0 },
+      sorting,
+    });
+  }, [
+    filterText,
+    filterType,
+    filterOperator,
+    data,
+    pagination,
+    sorting,
+    setTableURLState,
+  ]);
+
+  return (
+    <Box>
+      <HStack justifyContent="space-between">
+        <Box
+          alignItems="center"
+          border="1px solid"
+          borderColor="gray.800"
+          borderRadius="md"
+          color="gray.400"
+          display="flex"
+          gap={2}
+          p={1}
+          pl={2}
+          position="relative"
+        >
+          <FiFilter />
+          <Select.Root
+            collection={filterOptions}
+            onValueChange={(event) => setFilterType(event.value[0])}
+            size="xs"
+            value={[filterType ?? ""]}
+            width="90px"
+          >
+            <Select.Trigger>
+              <Select.ValueText placeholder="Filter by" />
+            </Select.Trigger>
+            <Select.Content>
+              {filterOptions.items.map((option) => (
+                <Select.Item item={option} key={option.value}>
+                  {option.label}
+                </Select.Item>
+              ))}
+            </Select.Content>
+          </Select.Root>
+          <Select.Root
+            collection={filterOperatorOptions}
+            onValueChange={(event) => setFilterOperator(event.value[0])}
+            size="xs"
+            value={[filterOperator ?? ""]}
+            width="140px"
+          >
+            <Select.Trigger>
+              <Select.ValueText placeholder="Filter operator" />
+            </Select.Trigger>
+            <Select.Content>
+              {filterOperatorOptions.items.map((option) => (
+                <Select.Item item={option} key={option.value}>
+                  {option.label}
+                </Select.Item>
+              ))}
+            </Select.Content>
+          </Select.Root>
+          <Input
+            border="0px"
+            onChange={(event) => setFilterText(event.target.value)}
+            outline="none"
+            placeholder="Filter text"
+            size="xs"
+            value={filterText}
+            width="200px"
+          />
+          {filterText === "" ? undefined : (
+            <CloseButton
+              aria-label="Clear search"
+              colorPalette="gray"
+              data-testid="clear-search"
+              onClick={() => {
+                setFilterText("");
+              }}
+              size="xs"
+            />
+          )}
+        </Box>
+
+        <HStack>
+          <Button colorPalette="blue">
```

Comment:
We can keep these buttons but let's set these as `disabled` and leave a TODO comment to connect them up later.